### DISCUSSION TITLE
what is the best way to build custom user interface with DX API V2?

### DISCUSSION THREAD

**Discussion by chaitanya-y** [pegasystems/react-sdk#333] [General]
what is the best way to build custom user interface with DX API V2 using react?

Should I use react starter pack or react-sdk?

**Comment by JEAxero** [comment]
The best way to learn about the difference between the Traditional DX API and the Constellation DX API (which is what I believe you're referring to as "V2") is to read the recently updated documentation and decide what's right for your situation.

https://community.pega.com/digital-experience-api

As shown on the pages linked from there, the Traditional Starter Packs (https://docs.pega.com/bundle/traditional-starter-packs/page/traditional-starter-packs/docs/traditional-starter-packs.html) which includes the React Starter Pack (https://community.pega.com/marketplace/component/react-starter-pack?) use the **Traditional DX API**.

If you want to use the **Constellation DX API** to build your own custom UI, you should use the **Constellation SDKs** (https://docs.pega.com/bundle/constellation-sdk/page/constellation-sdks/sdks/constellation-sdks.html).

Here's the detailed page that explains the differences between the **Traditional** and the **Constellation** DX APIs: https://docs.pega.com/bundle/dx-api/page/platform/dx-api/differences-between-v1-v2.html

Hope this helps!

**Comment by chaitanya-y** [comment]
can I use constellation design system with react material UI constellation-sdk? Is there any way to override material design with constellation design system for all the react sdk components?
